graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -44,6 +112,216 @@ def _are_filter_fields_uniquely_indexed(filter_fields, unique_indexes):
+                         u'exactly two parameter values: {} {}'
+                         .format(parameter_values, filter_operator))
+
+    lower_bound, upper_bound = None, None
+
+    if filter_operator == 'between':
+        # This (inclusively) constrains the lower and upper bounds of the
+        # values passing through the filter.
+        lower_bound = parameter_values[0]
+        upper_bound = parameter_values[1]
+    else:
+        raise AssertionError(u'Cost estimator found unsupported '
+                             u'ternary integer inequality operator {}.'
+                             .format(filter_operator))
+
+    query_interval = IntegerInterval(lower_bound, upper_bound)
+    if _is_IntegerInterval_empty(query_interval):
+        query_interval = None
+
+    return query_interval
+
+
+def _get_query_interval_of_integer_inequality_filter(parameter_values, filter_operator):
+    """Return IntegerInterval/None of values passing through a given integer inequality filter."""
+    if len(parameter_values) == 1:
+        query_interval = _get_query_interval_of_binary_integer_inequality_filter(
+            parameter_values, filter_operator
+        )
+    elif len(parameter_values) == 2:
+        query_interval = _get_query_interval_of_ternary_integer_inequality_filter(
+            parameter_values, filter_operator
+        )
+    else:
+        raise AssertionError(u'Cost estimator found filter operator {} with parameter values {}. '
+                             u'Currently, an operator must have either one or two parameter values.'
+                             .format(filter_operator, parameter_values))
+
+    return query_interval
+
+
+def _get_selectivity_of_integer_inequality_filter(
+    domain_interval, parameter_values, filter_operator
+):
+    """Return the selectivity of a given integer inequality filter filtering over a given interval.
+
+    First, we represent the filter being done in terms of the set of numbers that pass through the
+    given filter as an IntegerInterval. If a lower or upper bound does not exist, this is indicated
+    using None in the interval representation. For example, a '<' filter with a parameter value of 4
+    would be represented as (None, 4).
+    After this, we find the intersection of the domain being filtered and the filter interval.
+    The larger the intersection relative to the domain interval, the higher the selectivity. For
+    example, if the intersection is half as big as the domain interval, this will correspond to a
+    Fractional Selectivity with a selectivity factor of 0.5 (i.e. roughly 50% of elements pass
+    through the filter).
+
+    Args:
+        domain_interval: IntegerInterval namedtuple, describing the finite interval of integers
+                         being filtered.
+        parameter_values: List[int], describing the parameters for the inequality filter.
+        filter_operator: str, describing the inequality filter operation being performed.
```

Comment:
copy-paste the arg specs to all the helper functions. It really isn't clear what they mean just by their names.